src/main/java/net/snowflake/client/config/SFConnectionConfigParser.java

```diff
@@ -127,8 +124,15 @@ public static ConnectionParameters buildConnectionParameters() throws SnowflakeS
     }
   }
 
-  private static String createUrl(String account, Map<String, String> fileConnectionConfiguration) {
-    String host = String.format("%s.snowflakecomputing.com", account);
+  private static String createUrl(Map<String, String> fileConnectionConfiguration) {
+    String maybeAccount = fileConnectionConfiguration.get("account");
+    String maybeHost = fileConnectionConfiguration.get("host");
+    String host =
+        Optional.ofNullable(maybeHost)
+            .orElse(
+                Optional.ofNullable(maybeAccount)
+                    .map(acnt -> String.format("%s.snowflakecomputing.com", acnt))
+                    .orElse(null));
```

Comment:
In that case - should we throw an exception or something? At least log that no account/host provided?

Reply:
Added exception